docs/nebula-design-concepts.md

```diff
@@ -0,0 +1,216 @@
+# Data modeling and design concept
+------
+**Nebula Graph** is comprised of four components: storage layer, metadata service, computation layer and client console. In this documentation, we'll discuss data modeling and architecture of Nebula Graph.
+
+---
+## Directed property graph
+
+Modeling data with directed property graph, Nebula Graph is very easy to understand,
+i.e. Nebula Graph consists of two graph elements: vertex and edge. Each
+vertex represents an entity, and each edge represents how two vertices are associated.
+
+![image][example1]
+
+<!-- ### Vertex
+
+Vertices are typically used to store entities with properties. In the preceding example, vertices are **player** and **team**.
+
+### Edges
+
+Edges are used to connect vertices to one another and therefore provide a means
+of building your entities. Edges in Nebula Graph always have a type, a start (src)and an end vertex (dst), and a direction. It should be noted that there
+may be multiple edges of the same or different types at the same time between the
+start (src)and end vertex (dst). In the preceding example, edges are **serve**
+and **likeness**.
+
+### Properties
+
+Both vertices and edges can have properties, which are effectively key/value pairs. We call them **schema** in Nebula Graph. Like Mysql, Nebula Graph is a strong schema database. The name and data type of the properties are determined before the data is written. Vertex can have one or more properties, and edges can have properties too.
+
+In the preceding example, the schema of **player** has three kind of properties:
+
+- id (vid)
+- Name (string)
+- Age (int)
+
+ the schema of **serve** has two kinds of properties:
+
+ - start_year (int)
+ - end_year (int)
+
+### Tags
+
+In Nebula Graph, the type of vertex is called a tag. A vertex must have at least
+one tag, and Nebula allows you to set multiple tags per vertex.
+
+### Edgetype
+
+Each edge has an edgetype, and each edgetype defines the properties of the edge.
+这是按照我的理解方式写的
+-->
+
+
+### Vertex, type of vertex (tag) and the corresponding vertex properties
+
+In Nebula Graph, the type of vertex is called **tag**. **A vertex must have at least one type (tag)** or multiple types (tag). **Each tag has a corresponding set of properties called schema.**
+
+Consider the above picture as example, there are two types of vertices: **player** and **team**. Vertex player's schema has three properties: id (vid), Name(string), and Age(int); vertex team's schema has two properties: id (vid) and Name(string).
```

Comment:
In the example graph above, there are two tags: ** player** and **team**. _player_ has three properties: id(vid), name(string), and age(int); _team_ has two properties: id (vid) and Name(string).